onnxruntime/core/graph/graph.cc

```diff
@@ -64,12 +64,17 @@ static bool UsingLatestOnnxOpset(const DomainToVersionMap& opset_versions) {
 static Status MergeShapeInfo(const std::string& output_name,
                              const TypeProto_Tensor& source, TypeProto_Tensor& target,
                              bool strict, const logging::Logger& logger) {
-  try {
+  auto status = Status::OK();
+  ORT_TRY {
     ONNX_NAMESPACE::mergeInShapeInfo(source, target);
-  } catch (const ONNX_NAMESPACE::InferenceError& ex) {
+  }
+  ORT_CATCH(const ONNX_NAMESPACE::InferenceError& ex) {
     // if this model was not created with the latest onnx version, allow the shape inferencing failure (strict == false).
     // we do this to have strict testing of the latest inferencing to detect bugs, but lenient shape inferencing for
     // older models in case later changes to the ONNX shape inferencing or ORT break them.
+    ORT_UNUSED_PARAMETER(logger);
+    ORT_UNUSED_PARAMETER(strict);
+    ORT_UNUSED_PARAMETER(output_name);
     if (!strict) {
```

Comment:
Isn't 'strict' always used given it's in the 'if' condition?

Could you move the unused declarations for logger and output_name to the 'else' branch?